crates/provider/src/shared/provider.rs

```diff
@@ -0,0 +1,63 @@
+use std::error::Error;
+
+use async_trait::async_trait;
+
+use super::types::{NativeRunCommandOptions, PodDef, RunCommandResponse};
+
+#[async_trait]
+#[allow(non_upper_case_globals)]
+pub trait Provider {
+    async fn create_namespace(&mut self) -> Result<(), Box<dyn Error>>;
+    fn get_node_ip(&self) -> Result<String, Box<dyn Error>>;
```

Comment:
Same here, it can be relevant to use ```IpAddr``` from stdlib to encode our requirement inside the typesystem